static/js/src/advantage/react/components/Subscriptions/ExpiryNotification/ExpiryNotification.tsx

```diff
@@ -0,0 +1,132 @@
+import {
+  Notification,
+  NotificationSeverity,
+  PropsWithSpread,
+} from "@canonical/react-components";
+import type { NotificationProps } from "@canonical/react-components";
+import React, { ReactNode } from "react";
+
+import { UserSubscriptionStatuses } from "advantage/api/types";
+
+export enum ExpiryNotificationSize {
+  Small = "small",
+  Large = "large",
+}
+
+export enum StatusKey {
+  IsExpired = "is_expired",
+  IsInGracePeriod = "is_in_grace_period",
+  IsExpiring = "is_expiring",
+}
+
+type Props = PropsWithSpread<
+  {
+    multiple?: boolean;
+    size: ExpiryNotificationSize;
+    statuses: UserSubscriptionStatuses;
+  },
+  NotificationProps
+>;
+
+type Messages = Record<
+  StatusKey,
+  { [K in ExpiryNotificationSize]?: { message: ReactNode; title?: ReactNode } }
+>;
+
+// The expiry status keys in priority order.
+const STATUS_KEYS = [
+  StatusKey.IsExpired,
+  StatusKey.IsInGracePeriod,
+  StatusKey.IsExpiring,
+];
+
+/**
+ * A definition of the expiry notification messages.
+ * Large is displayed in the subscriptions header and details panel.
+ * Small is used in the subscription list.
+ */
+const MESSAGES: Messages = {
+  [StatusKey.IsExpiring]: {
+    [ExpiryNotificationSize.Large]: {
+      message:
+        'Select a subscription, then "Renew subscription..." to renew it.',
+      title: "Your subscription is about to expire.",
+    },
+    [ExpiryNotificationSize.Small]: {
+      message: "Subscription about to expire",
+    },
+  },
+  [StatusKey.IsExpired]: {
+    [ExpiryNotificationSize.Large]: {
+      message: (
+        <>
+          Please <a href="/contact-us">contact us</a>.
+        </>
+      ),
+      title: "This subscription has expired.",
+    },
+    [ExpiryNotificationSize.Small]: {
+      message: "Subscription expired",
+    },
+  },
+  [StatusKey.IsInGracePeriod]: {
+    [ExpiryNotificationSize.Large]: {
+      message: "It will be removed within 14 days if it's not renewed.",
+      title: "Your subscription has expired.",
+    },
+    [ExpiryNotificationSize.Small]: {
+      message: "Subscription expired",
+    },
+  },
+};
+
+const ExpiryNotification = ({
+  multiple = false,
```

Comment:
I noticed that the `is`/`has` naming convention for booleans is followed in multiple places at canonical, not only this repo, but also react-components. Might be a good idea to rename to `shouldAllowMultiple` or similar for consistency.

Reply:
Good idea. Done.